src/components/UserDetails/UserDetails.jsx

```diff
@@ -0,0 +1,219 @@
+import React, { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  Button,
+  Heading,
+  Stack,
+  Box,
+  HStack,
+} from '@chakra-ui/react';
+
+import styles from './UserDetails.module.css';
+import { passwordRequirementsRegex } from '../../utils/utils';
+import { getUserFromDB, updateFirebaseUser } from '../../utils/AuthUtils';
+
+const UserDetails = () => {
```

Comment:
to make this component reusable, add a `userId` parameter to fetch from the backend and display that user's information